app/models/game.rb

```diff
@@ -57,4 +57,30 @@ def create_pawns
       chess_pieces.create(type: 'Pawn', position_x: x, position_y: 7, color: :black)
     end
   end
+
+  def check?
+    king_is_in_check?('black') || king_is_in_check?('white')
+  end
+
+  private
+
+  def king_is_in_check?(color)
+    king = locate_king(color)
+    capturable_by_opposing_color?(king)
+  end
+
+  def locate_king(color)
+    chess_pieces.find_by(type: 'King', color: chess_pieces.colors[color])
+  end
+
+  def capturable_by_opposing_color?(king)
+    chess_pieces.where(color: chess_pieces.colors[opposite_color(king.color)]).find_each do |opponent|
+      return true if opponent.valid_move?(king.position_x, king.position_y)
+    end
+    false
+  end
+
+  def opposite_color(color)
```

Comment:
I agree - I think it should go up in `chess_piece` as the concept of finding the opposing color of the current piece seems like something piece-related as opposed to being specific to the check logic.

I think your solution is WAY more readable than mine, so I am happy... :+1: